### ISSUE TITLE
Inference error

### ISSUE THREAD

**Issue by spoonerj** [bmartacho/UniPose#14]
Dear Authors,

Thank you very much for your research. It is a very interesting research area!
I have recently adapted your code to very simply test on my own images, however, I think I have made some errors as my predictions are all wrong.

I have used the test function in `unipose.py` and uncommented line 165 of `utlis\utils.py` in the draw_paint function so that it could find the source image.

I have used the MPII args and downloaded the MPII pretrained weights for the inference. 

This is the source image I used, as you can see, it is a very simple human pose:
![test4](https://user-images.githubusercontent.com/43852330/111195217-f0c4c680-85b3-11eb-83cd-463d55e624d8.jpg)

and this is the output I get from the human pose estimation:

![0](https://user-images.githubusercontent.com/43852330/111195292-03d79680-85b4-11eb-8f17-530fef031e4c.png)

Can you please advise if you think there are any steps I might have done wrong?
I am happy to share my code if you want, but it is mainly adapted only to look for the correct files and that's it

I look forward to hearing from you

Many thanks

**Comment by GuaiYiHu** [comment]
Same issue as yours

**Comment by YuQi9797** [comment]
How you deal with the invisible points in the process of the experiment?

**Comment by nicolasugrinovic** [comment]
I have the same issue, output joints are not correct. This happens with MPII and LSP, both. I get worse results with the MPII weights. 
Here is a sample image from MPII dataset, the prediction is completely wrong and the heatmaps show nothing. 
![image](https://user-images.githubusercontent.com/25443240/113459556-32f46180-9416-11eb-9eda-cd4030f352ad.png)
![image](https://user-images.githubusercontent.com/25443240/113459569-3b4c9c80-9416-11eb-8718-4d0cf816b00e.png)

For LSP weights, the prediction is a little better but still wrong (the heatmaps look better than the previous). 
![image](https://user-images.githubusercontent.com/25443240/113459685-7e0e7480-9416-11eb-80c0-eb0babfbd3fe.png)

![image](https://user-images.githubusercontent.com/25443240/113459699-8797dc80-9416-11eb-9abb-e7c586282ab0.png)

**Comment by bmartacho** [comment]
Thanks for bringing the issue with the weights to our attention.

We uploaded two updated training weights that are better for generalization since the previous weights were use for testing and might have issues for outside use.

The weights can be found at the following link:
https://drive.google.com/drive/folders/1dPc7AayY2Pi3gjUURgozkuvlab5Vr-9n?usp=sharing

For the MPII dataset, the pre-trained weights are in the file UniPose_MPII.pth.
The best model for generalization outside the dataset in real images is the model trained on the COCO dataset (UniPose_COCO.pth) since it uses a more diverse and larger dataset.

**Comment by bmartacho** [comment]
> How you deal with the invisible points in the process of the experiment?

"Invisible" or occluded keypoints are still tentatively estimated by the network with the location where the method believes there is a higher likelihood of it occurring.

For the testing on the datasets used in this paper, occluded joints are detected but not counted for accuracy, as per pre-defined metrics by the dataset challenges.

In case you would like to not have the occluded joints in your visualization, one possible approach would be to set a minimum confidence threshold for the detected joints. I.e. only show joints with detection confidence >= 0.5.

**Comment by EveningLin** [comment]
I use the tar file, because the pth file will report errors, and the joint points are basically all wrong. What should I do to get better joint points based on MPI
![1](https://user-images.githubusercontent.com/110521494/211269560-7f204b7a-4ba7-478e-a0a6-9baae2251255.png)
I
